elements/lisk-p2p/src/peer_selection.ts

```diff
@@ -63,8 +63,23 @@ export const selectPeersForRequest = (
 
 export const selectPeersForSend = (
 	input: P2PPeerSelectionForSendInput,
-): ReadonlyArray<P2PDiscoveredPeerInfo> =>
-	shuffle(input.peers).slice(0, input.peerLimit);
+): ReadonlyArray<P2PDiscoveredPeerInfo> => {
```

Comment:
What happens if there only 10 outbound connections? even though we have more inbound peers that we can send to, we will still send to (5 + 12) peers.
Do you think we should check at the end if we have all 24 peers to send? and maybe select more peers from other `kind` to fill it up.

Reply:
Good point. I guess we can add a step before this logic to check if `input.peers.length` is less than `input.peerLimit`; if so, then we just return all of them.

Reply:
In that case, it would send to (10 + 12 peers) because `halfPeerLimit` will be 12. Still 2 short of 24 though so I agree we should pad the list with peers from the other kind.